src/scripts/themed_posts.js

```diff
@@ -24,41 +29,63 @@ const processPosts = async function (postElements) {
   filterPostElements(postElements, { includeFiltered: true }).forEach(async postElement => {
     if (postElement.matches(blogViewSelector) && !enableOnPeepr) return;
 
-    const { blog: { name, theme } } = await timelineObject(postElement);
-
-    if (blacklist.includes(name)) return;
-
-    if (!blogs.has(name)) {
-      blogs.add(name);
-
-      const {
-        backgroundColor,
-        titleColor,
-        linkColor
-      } = theme;
-
-      const backgroundColorRGB = hexToRGB(backgroundColor);
-      const titleColorRGB = hexToRGB(titleColor);
-      const linkColorRGB = hexToRGB(linkColor);
-
-      styleElement.textContent += `
-        [data-xkit-themed="${name}"] {
-          --white: ${backgroundColorRGB};
-          --black: ${titleColorRGB};
-          --accent: ${linkColorRGB};
-          --color-primary-link: rgb(var(--accent));
-        }
-      `;
+    const { blog, trail = [] } = await timelineObject(postElement);
+
+    const blogData = [
+      blog,
+      ...reblogTrailTheming ? trail.map(item => item.blog) : []
+    ];
+
+    blogData.forEach(({ name, theme }) => {
+      if (blacklist.includes(name)) return;
+
+      if (!blogs.has(name)) {
+        blogs.add(name);
+
+        const {
+          backgroundColor,
+          titleColor,
+          linkColor
+        } = theme;
+
+        const backgroundColorRGB = hexToRGB(backgroundColor);
+        const titleColorRGB = hexToRGB(titleColor);
+        const linkColorRGB = hexToRGB(linkColor);
+
+        styleElement.textContent += `
+          [data-xkit-themed="${name}"] {
+            --white: ${backgroundColorRGB};
+            --black: ${titleColorRGB};
+            --accent: ${linkColorRGB};
+            --color-primary-link: rgb(var(--accent));
+          }
+        `;
+      }
+    });
+
+    postElement.dataset.xkitThemed = blog.name ?? '';
+
+    if (reblogTrailTheming) {
+      [...postElement.querySelectorAll(reblogSelector)].forEach((reblog, i) => {
+        reblog.dataset.xkitThemed = trail[i]?.blog?.name ?? '';
+      });
     }
-
-    postElement.dataset.xkitThemed = name;
   });
 };
 
 export const main = async function () {
-  ({ enableOnPeepr, blacklistedUsernames } = await getPreferences('themed_posts'));
+  ({ reblogTrailTheming, enableOnPeepr, blacklistedUsernames } = await getPreferences('themed_posts'));
   blacklist = blacklistedUsernames.split(',').map(username => username.trim());
 
+  reblogSelector = await keyToCss('reblog');
+
+  styleElement.textContent += `
+    article ${reblogSelector} {
+      margin-top: 0;
+      padding-bottom: 1px;
+    }
```

Comment:
Here's that forbidden knowledge you ordered:

```suggestion
    article ${reblogSelector} {
      overflow-y: hidden;
      margin-top: 0;
    }

    article ${reblogSelector}:not(:last-child) > :last-child {
      margin-bottom: 15px;
    }
```

Setting `overflow: hidden` makes descendant margins count towards the content box size, instead of collapsing through it. Descendant margins can still collapse into _each other_ though, allowing us to set a minimum amount of bottom spacing without padding at all.

Reply:
I think I have read this paragraph like 10 times :D I think I get most of it. Out of curiosity, Is this related to this paragraph?

> Specifying a value other than visible (the default) or clip creates a new [block formatting context](https://developer.mozilla.org/en-US/docs/Web/Guide/CSS/Block_formatting_context). This is necessary for technical reasons — if a float intersected with the scrolling element it would forcibly rewrap the content after each scroll step, leading to a slow scrolling experience.
https://developer.mozilla.org/en-US/docs/Web/CSS/overflow#description